### PR TITLE
Parse XTen Flowcells

### PR DESCRIPTION
**Pull request by mariogiov**
Alright this will work with the format we have decided upon for XTen flowcells. I've tested it with `141229_ST-E00214_0019_AH00V3CCXX` but if there are any from Uppsala it would be nice to test with those as well.

### CODE REVIEW

File: ngi_pipeline/conductor/flowcell.py [hunk 1]
```diff
@@ -159,11 +157,11 @@ def setup_analysis_directory_structure(fc_dir, projects_to_analyze,
         return []
     # Map the directory structure for this flowcell
     try:
-        fc_dir_structure = parse_casava_directory(fc_dir)
+        fc_dir_structure = parse_xten_directory(fc_dir)
```
Comment (vezzi) [inline ngi_pipeline/conductor/flowcell.py]: @mariogiov can we rename the functions to better names: "parse_bcl2fastq1_8_directory" and "parse_bcl2fastq2_5_directory"?

File: ngi_pipeline/conductor/flowcell.py [hunk 2]
```diff
@@ -249,38 +232,21 @@ def setup_analysis_directory_structure(fc_dir, projects_to_analyze,
...
-                                                                            fc_full_id,
-                                                                            fq_file))
-                        libprep_name = "Unknown"
+                    LOG.warn('Project "{}" / sample "{}" / seqrun "{}" / fastq "{}" '
+                             'has no libprep information in Charon. Setting '
+                             'library prep to "Unknown"'.format(project_name,
+                                                                sample_name,
+                                                                fc_full_id,
+                                                                fq_file))
+                    libprep_name = "Unknown"
```
Comment (vezzi) [inline ngi_pipeline/conductor/flowcell.py]: Much simpler now :-D

File: ngi_pipeline/conductor/flowcell.py [hunk 3]
```diff
@@ -411,20 +375,20 @@ def parse_casava_directory(fc_dir):
...
     │   ├── J_Lundeberg_14_24
-    │   │   ├── P1775_112
-    │   │   ├── P1775_120
-    │   │   ├── P1775_128
-    │   │   ├── P1775_136
-    │   │   ├── P1775_144
-    │   │   ├── P1775_152
-    │   │   ├── P1775_160
-    │   │   └── P1775_168
+    │   │   ├── Sample_P1775_112
```
Comment (vezzi) [inline ngi_pipeline/conductor/flowcell.py]: My bad!!! Sample is not necessary and I wanted to get rid of it but I pointed you to an example where sample was still there.
The Adults version of the flowcell "141229_ST-E00214_0019_AH00XXXXXX" is without "Sample" and it is like this that should be (as we agree with Patrick)

File: ngi_pipeline/conductor/flowcell.py [hunk 4]
```diff
@@ -438,41 +402,42 @@ def parse_xten_directory(fc_dir):
...
-
+        if STHLM_PROJECT_RE.match(project_name):
+            # We need to change J_Lundeberg_15_01 to J.Lundeberg_15_01
+            project_name = project_name.replace('_', '.', 1)
         # e.g. <Project_dir>/P680_356F_dual56/
         for sample_dir in glob.glob(sample_dir_pattern):
             LOG.info('Parsing samples directory "{}"...'.format(sample_dir.split(
                                                 os.path.split(fc_dir)[0] + "/")[1]))
-            sample_name = os.path.basename(sample_dir)
+            sample_name = os.path.basename(sample_dir).replace('Sample_', '', 1)
```
Comment (vezzi) [inline ngi_pipeline/conductor/flowcell.py]: this is what I wanted to avoid

**Comment by vezzi** [comment]
@mariogiov fine, the only "error" is the "Sample_" in the sample name... sorry my fault :-D

**Comment by mariogiov** [comment]
Alright I made those changes -- should be good now.

**Pull request merged**